### ISSUE TITLE
Proposal: Split up matchmaker_config.yaml

### ISSUE THREAD

**Issue by jeremyje** [googleforgames/open-match#130]
The matchmaker_config.yaml has grown to be a very large file and Open Match is a set up different servers with different context. We should consider splitting up the configuration so that each server is only provided the context necessary to run properly.

This should improve reliability because it'll force users to have multiple ConfigMaps for each configuration so if there's a bad config change it'll only impact a subset of server types instead of all of them.

**Comment by ihrankouski** [comment]
I believe the ConfigMap could still remain single, but instead it will contain several entries in "data" section. Some pieces of configuration should be shared by services, some are independent, isn't it?

**Comment by joeholley** [comment]
Thought about this for a bit.  There are definitely some per-component configs, where every component can have different values for these without issues. In fact, in many situations it would be preferred.  In particular, having the logging config and debug flags on a 'per component' basis would be nice.  

However, the majority of the parameters in the configs are there specifically for sharing - so that the values aren't baked into the code and when a change needs to be made, every component gets the same values (and hopefully avoid a situation where multiple components could accidentally have different values). To be honest, it would probably be a good idea to isolate the ones of those that we don't want beginners accidentally changing, like service names  and port numbers.  I still wouldn't want them baked directly into the code, but maybe move them back to being in a config location built into the container images by default.  We could then update the OM config loading module to 'override' the defaults with values from a configmap if it exists.  That way expert users (people deploying OM to a K8s cluster with many other apps and potential port conflicts, for example) could configure those sorts of things, but beginners aren't exposed to all that extra complexity that they probably don't need to mess with at all.  Some other things, like the list of properties to index, need to be kept in sync between the component writing the index (FE API) and the component deleting the index (BE API), but also should absolutely be modified by someone coding a matchmaker against OM.

Right now based on 0.4.0 WIP branch, and leaving out things that are getting deprecated and should be removed in future versions:

I see the first group ('per component' config) as containing:
```
debug: true
logging:
  level: debug
  format: text
  source: true
redis:
  pool:
    maxIdle: 3
    maxActive: 0
    idleTimeout: 60
  queryArgs:
    count: 10000
  results:
    pageSize: 10000
```
The second group ('we should hide this but allow overrides for experts' config):
```
api: 
  backend: 
    hostname: om-backendapi
    port: 50505
  frontend: 
    hostname: om-frontendapi
    port: 50504
  mmlogic: 
    hostname: om-mmlogicapi
    port: 50503
  functions:
    port: 50502
metrics: 
  port: 9555
  endpoint: /metrics
queues: 
  proposals: 
    name: proposalq
ignoreLists: 
  proposed: 
    name: proposed
  deindexed: 
    name: deindexed
  expired: 
    name: OM_METADATA.accessed
```
And the third ('expose to users but any change should go to all components by default' config):
```
api: 
  backend: 
    backoff: "[2 32] *2 ~0.33 <30"
  frontend: 
    backoff: "[2 32] *2 ~0.33 <300"
evalutor: 
  intervalMs: 10000
  pollIntervalMs: 1000
metrics: 
  reportingPeriod: 5
ignoreLists: 
  proposed: 
    offset: 0
    duration: 800 
  deindexed: 
    offset: 0
    duration: 800 
  expired: 
    offset: 800 
    duration: 0 
redis:
  expirations:
    player: 43200
    matchobject: 43200
playerIndices:
- char.cleric
- char.knight
- char.paladin
- map.aleroth
- map.oasis
- mmr.rating
- mode.battleroyale
- mode.ctf
- mode.demo
- region.europe-east1
- region.europe-west1
- region.europe-west2
- region.europe-west3
- region.europe-west4
- role.dps
- role.support
- role.tank
```

**Comment by joeholley** [comment]
Note, if we did split these out into these categories, it would probably be a good idea to change some of the hierarchies, which don't make as much sense if not all the values are in the same config.

**Comment by ihrankouski** [comment]
Anybody working on this? I'd like to take it.

**Comment by jeremyje** [comment]
I can't seem to assign this issue to you.

**Comment by sawagh** [comment]
We have revisiting the configuration for Open Match on v0.7 roadmap (exploring ways to change  application level configuration such as indexes more dynamically). This task will likely merge into that work. Not in scope for v0.6

**Comment by yfei1** [comment]
Closed by #652 and #667
